plutus-core/plutus-core/src/PlutusCore/Constant/Typed.hs

```diff
@@ -521,6 +526,46 @@ instance (HasConstantIn uni term, GShow uni, GEq uni, uni `Contains` a) =>
 -- | A constraint for \"@a@ is a 'KnownType' by means of being included in @UniOf term@\".
 type KnownBuiltinType term a = KnownBuiltinTypeIn (UniOf term) term a
 
+{- Note [Performance of KnownTypeIn instances]
+Even though we don't use 'makeKnown' and 'readKnown' directly over concrete types, it's still
+beneficial to inline them, because otherwise GHC compiles each of them to two definitions
+(one calling the other) for some reason.
+
+Similarly, we inline implementations of 'knownTypeAst' just to get tidier Core.
```

Comment:
Should that be `toTypeAst`, or maybe `KnownTypeAst`?

Reply:
Yes, thank you.